common/src/main/java/tachyon/client/FileSystemMasterClient.java

```diff
@@ -102,16 +102,15 @@ public synchronized long getFileId(String path) throws IOException {
    * @return the file info for the given file id
    * @throws IOException if an I/O error occurs
    */
-  public synchronized FileInfo getFileInfo(long fileId) throws IOException {
+  public synchronized FileInfo getFileInfo(long fileId) throws IOException,
+      FileDoesNotExistException {
     int retry = 0;
     while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
       connect();
       try {
         return mClient.getFileInfo(fileId);
       } catch (FileDoesNotExistException e) {
-        throw new IOException(e);
-      } catch (InvalidPathException e) {
-        throw new IOException(e);
+        throw e;
```

Comment:
Same here and for the rest of this PR, unless you want add additional logic, e.g. logging, you can remove this catch clause.